ntoskrnl/config/cmwraprs.c

```diff
@@ -114,9 +144,35 @@ CmpFileWrite(IN PHHIVE RegistryHive,
     if (CmpNoWrite)
         return TRUE;
 
+    /* Create the event to wait for I/O completion */
+    Status = CmpCreateEvent(SynchronizationEvent, &EventHandle, &Event);
```

Comment:
Since this even is created for each write operation (!!), isn't it possible to have it created once (per hive file) and be reused? :thinking: